pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -196,7 +198,22 @@ func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOpera
 		}
 
 		var updated bool
+		var mcoOperatorImageUpgrading bool
+		if operator.Name == "machine-config" {
+			for _, version := range operator.Status.Versions {
+				if version.Name == "operator-image" {
+					if targetImagePullSpec := getMCOImagePullSpec(mcoDeployment); targetImagePullSpec != version.Version {
+						mcoOperatorImageUpgrading = true
+						break
+					}
+				}
+			}
+		}
+
 		for _, version := range operator.Status.Versions {
+			if operator.Name == "machine-config" && mcoOperatorImageUpgrading {
```

Comment:
nit: this conditional doesn't depend on the `Versions` you're iterating over, so you could pull it outside the loop:

```golang
if operator.Name != "machine-config" || !mcoOperatorImageUpgrading {
  for _, version := range operator.Status.Versions {
    if version.Name == "operator" && version.Version == targetVersion {
      updated = true
      break
    }
  }
}
```

/hold in case you want to refactor like that

Reply:
Done.